baseplate/lib/secrets.py

```diff
@@ -495,8 +495,11 @@ def __init__(
         backoff: Optional[float] = None,
     ):  # pylint: disable=super-init-not-called
         self.path = Path(path)
+        if not self.path.is_dir():
+            raise ValueError(f"{self.path} must be a directory")
         self.cache = {}
         self.data_symlink = self.path.joinpath("..data")
```

Comment:
Can we check if this exists instead of the dir? That will also detect using a subdir of the CSI volume

We may need a filewatcher for this but I assume and HOPE the volume is fully populated before the pod gets it mounted.

Reply:
I'm nearly certain the volume will exist before the container we care about starts running. I can't find the docs for it right now, but there's something about how these mounts are loaded earlier in the kubernetes lifecycle and pods will fail to start if they can't properly mount

Reply:
Yea, that's what I'm seeing in snoodev. This is something I can test myself as well.